src/hosted-buttons/utils.js

```diff
@@ -77,6 +85,33 @@ export const createAccessToken: CreateAccessToken = memoize<CreateAccessToken>(
   }
 );
 
+export const getButtonPreferences = ({
+  button_preferences: buttonPreferences,
+  eligible_funding_methods: eligibleFundingMethods,
+}: NcpResponsePreferences): HostedButtonPreferences => {
+  if (!buttonPreferences?.length || !eligibleFundingMethods?.length) {
+    throw new Error(
+      `Expected preferences to be populated, recieved: ${JSON.stringify({
+        buttonPreferences,
+        eligibleFundingMethods,
+      })}`
+    );
+  }
+
+  return {
+    // Remove any buttons that are not included in eligibleFundingMethods
+    buttonPreferences: buttonPreferences.filter(
+      (fundingMethod) =>
+        eligibleFundingMethods.includes(fundingMethod) ||
+        fundingMethod === "default"
+    ),
+    // Sort the eligible funding methods returned from /ncp/api/form-fields in the order that they would appear in the smart stack.
+    eligibleFundingMethods: SUPPORTED_FUNDING_SOURCES.filter((fundingMethod) =>
+      eligibleFundingMethods.includes(fundingMethod)
+    ),
```

Comment:
`SUPPORTED_FUNDING_SOURCES` is an array we use to determine the order of the buttons for the smart stack.  Doing the same thing here, but filtering it down to only the buttons that are eligible.